sys/net/gnrc/netif/gnrc_netif.c

```diff
@@ -720,9 +720,27 @@ gnrc_netif_t *gnrc_netif_get_by_prefix(const ipv6_addr_t *prefix)
 int gnrc_netif_ipv6_group_join_internal(gnrc_netif_t *netif,
                                         const ipv6_addr_t *addr)
 {
-    unsigned idx = UINT_MAX;
+    uint8_t group_data[GNRC_NETIF_L2ADDR_MAXLEN];
+    unsigned idx;
 
+    /* can be called out of lock */
+    idx = gnrc_netif_ipv6_group_to_l2_group(netif, addr, group_data);
     gnrc_netif_acquire(netif);
+    if (idx > 0) {
+        gnrc_netapi_opt_t group = {
+            .opt = NETOPT_L2_GROUP,
+            .data = group_data,
+            .data_len = (uint16_t)idx;
+        }
+        idx = netif->ops->set(netif, &group);
+        if ((idx < 0) && (idx != -ENOTSUP)) {
```

Comment:
`idx` is declared as `unsigned`, so it can never be below `0` and therefore this expression will always be false. 

Also `idx` is compared to an signed value (`-ENOTSUP`).